### PR TITLE
Allow throw in SYSCALLs

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by shargon** [comment]
Now that we consume the stack items in syscalls, it's safe to catch the exception, right?

**Comment by erikzhang** [comment]
Yes.

File: src/neo-vm/ExecutionEngine.cs [hunk 1]
```diff
@@ -340,8 +340,7 @@ private void ExecuteInstruction(Instruction instruction)
                     }
                 case OpCode.THROW:
                     {
-                        UncaughtException = Pop();
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs:343]: How the syscall will call this opcode, when syscall throw an exception it goes to `OnFault`

Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs:343]: They should call the method `Throw()`.

https://github.com/neo-project/neo-vm/blob/98cde1d9eb920033d39dec39c13bba23cb76090a/src/neo-vm/ExecutionEngine.cs#L1451

**Pull request merged**